src/lib.nix

```diff
@@ -94,6 +94,8 @@
       dream2nixConfig = config;
       callPackageDream =
         throw "callPackageDream is not available before nixpkgs is imported";
+      pkgs = throw "pkgs is not available before nixpkgs is imported";
+      utils = throw "utils is not available before nixpkgs is imported";
```

Comment:
We could but I don't think we necessarily need to since lazy eval... But we may still want to do it anyways for the sake of organization and making it clear which ones need nixpkgs and which ones don't. I did it like this so we can clean it up later, it shouldn't be complex to remove the pieces, should just be a matter of creating a new `top-level.nix` that only imports ones that need nixpkgs and another for ones that don't and respective `framework.nix` files. What do you think about that?